examples/MNIST_classification.ipynb

```diff
@@ -0,0 +1,598 @@
+    "                    for direction in direction_list]\n",
+    "\n",
+    "steps_list = [ [\n",
+    "    ('binarizer', Binarizer(threshold=0.4)),\n",
+    "    ('filtration', filtration),\n",
+    "    ('diagram', CubicalPersistence()),\n",
+    "    ('amplitude', Amplitude(metric='heat', metric_params={'p': 2}))]\n",
+    "    for filtration in filtration_list ]\n",
+    "\n",
+    "pipeline_list = [ (str(direction_list[i]), Pipeline(steps_list[i])) for i in range(len(steps_list))]\n",
+    "feature_union_filtrations = FeatureUnion(pipeline_list, n_jobs=-1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "feature_union_filtrations.fit(X_train[:20])\n",
+    "X_train_filtrations = feature_union_filtrations.transform(X_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Deriving a full-scale TDA feature extraction pipeline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can go full-scale and extract a large number of features. Careful, some of them will be highly correlated, so it would be good to use a feature selection algorithm to reduce their number before passing them to a classifier."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "direction_list = [ [1, 0], [1, 1], [0, 1], [-1, 1], [-1, 0], [-1, -1], [0, -1], [1, -1] ] \n",
+    "center_list = [ [13, 6], [6, 13], [13, 13], [20, 13], [13, 20], [6, 6], [6, 20], [20, 6], [20, 20] ]\n",
+    "n_iterations_erosion_list = [6, 10]\n",
+    "n_iterations_dilation_list = [6, 10]\n",
+    "n_iterations_signed_list = [6, 10]\n",
+    "n_neighbors_list = [2, 4]\n",
+    "\n",
+    "# Creating a list of all filtration transformer, we will be applying\n",
+    "filtration_list =  [HeightFiltration(direction=np.array(direction)) \n",
+    "                    for direction in direction_list] \\\n",
+    "                 + [RadialFiltration(center=np.array(center)) \n",
+    "                    for center in center_list] \\\n",
+    "                 + [ErosionFiltration(n_iterations=n_iterations) \n",
+    "                    for n_iterations in n_iterations_erosion_list] \\\n",
+    "                 + [DilationFiltration(n_iterations=n_iterations) \n",
+    "                    for n_iterations in n_iterations_dilation_list] \\\n",
+    "                 + [SignedDistanceFiltration(n_iterations=n_iterations) \n",
+    "                    for n_iterations in n_iterations_signed_list] \\\n",
+    "                 + ['passthrough']\n",
+    "\n",
+    "# Creating the diagram generation pipeline\n",
+    "diagram_steps = [[Binarizer(threshold=0.4), \n",
+    "                  filtration, \n",
+    "                  CubicalPersistence(homology_dimensions=[0, 1]), \n",
+    "                  Scaler(metric='bottleneck')] \n",
```

Comment:
I find it a little strange that the scaler alone improves results substantially. Normally, I'd expect a scaler to be followed by a filter, but if it's not then can't the model weight take care of the different scales between homology dimensions?